lib/thrift/parser/models.ex

```diff
@@ -61,17 +61,14 @@ defmodule Thrift.Parser.Models do
 
     @spec new(charlist, %{charlist => enum_value}) :: t
     def new(name, values) do
-      values = values
-      |> Enum.with_index
-      |> Enum.map(fn
-        {{name, value}, _index} ->
-          {atomify(name), value}
-
-        {name, index} ->
-          {atomify(name), index + 1}
+      {_, values} = Enum.reduce(values, {1, []}, fn
```

Comment:
Should the initial index start at 0 instead of 1?

Apache Thrift [initializes this value to -1](https://github.com/apache/thrift/blob/master/compiler/cpp/src/thrift/thrifty.yy#L488) and [uses a pre-increment](https://github.com/apache/thrift/blob/master/compiler/cpp/src/thrift/thrifty.yy#L528) to assign each implicit value. ([Explicit values are taken literally.](https://github.com/apache/thrift/blob/master/compiler/cpp/src/thrift/thrifty.yy#L517))

The result is that an enumeration with only implicit values starts at 0.

Reply:
You're absolutely right. I tested generating some enums and it indeed starts at 0. I hadn't checked, I was just replicating the functionality here that already started at 1.
Turns out this is a substantially bigger bug than I anticipated, all our enums had the potential to be off by one.